lib/academic_history/pdf_processor.rb

```diff
@@ -0,0 +1,46 @@
+require 'pdf-reader'
+
+module AcademicHistory
+  class PdfProcessor
+    include Enumerable
+
+    AcademicEntry = Struct.new(:name, :credits, :number_of_failures, :date_of_completion, :grade) do
+      def approved?
+        grade != '***'
+      end
+    end
+
+    def initialize(file)
+      @reader = PDF::Reader.new(file.path)
+    end
+
+    def each
+      reader.pages.each do |page|
+        page.text.split("\n").each do |line|
```

Comment:
```suggestion
        page.text.each_line do |line|
```

no es exactamente lo mismo porque cada línea va a terminar con `/n`, pero creo que en este caso te sirve igual